src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -367,8 +367,9 @@ Klass* SystemDictionary::resolve_array_class_or_null(Symbol* class_name,
       k = k->array_klass(ndims, CHECK_NULL);
     }
   } else {
-    k = Universe::typeArrayKlass(t);
-    k = TypeArrayKlass::cast(k)->array_klass(ndims, CHECK_NULL);
+    TypeArrayKlass* tak = Universe::typeArrayKlass(t);
+    k = tak->array_klass(ndims, CHECK_NULL);
+    k = k->array_klass(ndims, CHECK_NULL);
```

Comment:
this looks puzzling.  Why are there two array_klass calls now?  Add short comments to explain.

Reply:
I sort of see now but am getting squint lines.  It's not important for performance to eliminate a virtual call here.

Reply:
Hmm. Having two `array_klass` calls were not intentional. I accepted Dean's suggestion in the GitHub UI, but that didn't remove the old `array_klass`. I think I'll revert that change given that it is not important to devirtualize this.

Reply:
Why devirtualize elsehwere but not here?  Maybe it's not a big deal.

Reply:
Or to put it another way, what's the advantage of using `TypeArrayKlass* tak` in similar situations below?